manifests/sentinel.pp

```diff
@@ -106,34 +106,34 @@
 #
 class redis::sentinel (
   $auth_pass              = undef,
-  $config_file            = $redis::params::sentinel_config_file,
-  $config_file_orig       = $redis::params::sentinel_config_file_orig,
+  Stdlib::Absolutepath $config_file = $redis::params::sentinel_config_file,
+  Stdlib::Absolutepath $config_file_orig = $redis::params::sentinel_config_file_orig,
   Stdlib::Filemode $config_file_mode = '0644',
   String[1] $conf_template = 'redis/redis-sentinel.conf.erb',
-  $daemonize              = $redis::params::sentinel_daemonize,
-  $down_after             = 30000,
-  $failover_timeout       = 180000,
-  $init_script            = $redis::params::sentinel_init_script,
+  Boolean $daemonize = $redis::params::sentinel_daemonize,
+  Integer[0] $down_after = 30000,
+  Integer[0] $failover_timeout = 180000,
+  Optional[Stdlib::Absolutepath] $init_script = $redis::params::sentinel_init_script,
   String[1] $init_template = 'redis/redis-sentinel.init.erb',
-  $log_level              = $redis::params::log_level,
-  $log_file               = $redis::params::log_file,
+  String[1] $log_level = 'notice',
+  Stdlib::Absolutepath $log_file = '/var/log/redis/redis.log',
   String[1] $master_name  = 'mymaster',
   Stdlib::Host $redis_host = '127.0.0.1',
-  Stdlib::Port $redis_port = $redis::params::port,
-  $package_name           = $redis::params::sentinel_package_name,
+  Stdlib::Port $redis_port = 6379,
+  String[1] $package_name = $redis::params::sentinel_package_name,
   String[1] $package_ensure = 'present',
   Integer[0] $parallel_sync = 1,
   Stdlib::Absolutepath $pid_file = '/var/run/redis/redis-sentinel.pid',
-  Integer[0] $quorum      = 2,
-  $sentinel_bind          = undef,
+  Integer[0] $quorum = 2,
+  $sentinel_bind = undef,
   Stdlib::Port $sentinel_port = 26379,
-  $service_group          = $redis::params::service_group,
+  String[1] $service_group = $redis::params::service_group,
   String[1] $service_name = 'redis-sentinel',
-  $service_ensure         = $redis::params::service_ensure,
-  Boolean $service_enable = $redis::params::service_enable,
-  $service_user           = $redis::params::service_user,
+  Stdlib::Ensure::Service $service_ensure = 'running',
+  Boolean $service_enable = true,
+  String[1] $service_user = 'redis',
   Stdlib::Absolutepath $working_dir = '/tmp',
-  $notification_script    = undef,
+  $notification_script = undef,
```

Comment:
The (unit) test uses bar.sh. I don't know if that's actually valid or just an example. The Redis docs aren't explicit about it. Do you know?

Reply:
@ekohl Looking at codebase and docs I couldn't find anything useful as well.
Looking at all examples people were using it online, it was always full path to script.

I think limit it to absolute path is not a bad idea, regardless whether bar.sh would be also accepted.
It will also resolve confusion for other people, and the question where to put this bar.sh to make it work.

What do you think ?

Reply:
I agree. It's also easier to go from an absolute path to a string since that's non-breaking.